.travis.yml

```diff
@@ -134,15 +136,12 @@ install:
     set +e
 
 script:
-  - ./bootstrap
-  - ./configure $FFCONFIG
-  - make -j4
-  - make -C contrib/fonttools -j4
-  - make install
-  - if [ ! -z "$TRACE_CODE_COVERAGE" ]; then make check ; coveralls --gcov-options '\-bulp' --gcov $(which $GCOV) ; fi
-  - if [ "$TRAVIS_OS_NAME" == "linux" ]; then make distcheck -j4; fi
-  - fontforge -version
-  - $PYTHON -c "import fontforge; import psMat; print(fontforge.__version__, fontforge.version()); fontforge.font();"
+  - if [ -n "$MODE" ] ; then ./bootstrap && ./configure $FFCONFIG ; fi
+  - if [ "$MODE" == "compilation" ] || [ "$MODE" == "coverage" ] ; then make && make -C contrib/fonttools && make install ; fi
+  - if [ "$MODE" == "coverage" ] ; then make check ; coveralls --gcov-options '\-bulp' --gcov $(which $GCOV) ; fi
+  - if [ "$MODE" == "distcheck" ] ; then make distcheck ; fi
```

Comment:
Doing a distcheck build separate from 'compilation' is pointless, because to do a distcheck, it needs to run the build first anyway.

Reply:
@jtanx? distcheck does not need nor does run the build first...

The first thing it does is creating the dist tarball, then untar it and only then run the full compilation cycle inside the release directory. Only `./configure` and `make check` is run twice.

You can check that no build is done before creating the tarball in the logs of one of those distcheck-only builds: <https://api.travis-ci.org/jobs/283364498/log.txt> and <https://travis-ci.org/fontforge/fontforge/jobs/283364498>.

Removing distcheck from the main build allows the main build to pass/fail faster. Same for the distcheck build itself.